src/Std/Data/DHashMap/Raw.lean

```diff
@@ -0,0 +1,438 @@
+/-
+Copyright (c) 2024 Lean FRO, LLC. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Markus Himmel
+-/
+prelude
+import Init.Data.BEq
+import Init.Data.Hashable
+import Std.Data.DHashMap.Internal.Defs
+
+/-
+# Dependent hash maps with unbundled well-formedness invariant
+
+This file develops the type `Std.Data.DHashMap.Raw` of dependent hash
+maps with unbundled well-formedness invariant.
+
+This version is safe to use in nested inductive types. The well-formedness predicate is
+available as `Std.Data.DHashMap.Raw.WF` and we prove in this file that all operations preserve
+well-formedness. When in doubt, prefer `DHashMap` over `DHashMap.Raw`.
+
+Lemmas about the operations on `Std.Data.DHashMap.Raw` are available in the module
+`Std.Data.DHashMap.RawLemmas`.
+-/
+
+set_option linter.missingDocs true
+set_option autoImplicit false
+
+universe u v w
+
+variable {α : Type u} {β : α → Type v} {δ : Type w} {m : Type w → Type w} [Monad m]
+
+namespace Std
+
+open DHashMap.Internal DHashMap.Internal.List
+
+namespace DHashMap
+
+namespace Raw
+
+/-!
+The type `DHashMap.Raw` itself is defined in the module `Std.Data.DHashmap.RawDef` for import
+structure reasons.
+-/
+
+/-- Creates a new empty hash map. The optional parameter `capacity` can be supplied to presize the
+map so that it can hold the given number of mappings without reallocating. It is also possible to
+use the empty collection notations `∅` and `{}` to create an empty hash map with the default
+capacity. -/
+@[inline] def empty (capacity := 8) : Raw α β :=
+  (Raw₀.empty capacity).1
+
+instance : EmptyCollection (Raw α β) where
+  emptyCollection := empty
+
+/-- Insert the given mapping into the map, replacing an existing mapping for the key if there is
+one. -/
```

Comment:
```suggestion
/--
Inserts the given mapping into the map, replacing an existing mapping for the key if there is
one.
-/
```

Indicative mood, plus the format we decided on for multi-line docstring comment indicator placement

Reply:
Oh no, for some reason I remembered that we settled on the exact opposite convention. I should have double-checked. I will go over all of them and change them.

Reply:
Gotta get that linter :-)